pkg/providers/amazon/amazon.go

```
@@ -0,0 +1,77 @@
+package amazon
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-node-termination-handler/pkg/ec2metadata"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/google/go-containerregistry/pkg/authn"
+)
+
+type Provider struct {
+	ecrClient       *ecr.Client
+	authToken       *authn.AuthConfig
+	authTokenExpiry time.Time
+}
+
+func NewProvider() *Provider {
+	cfg, _ := config.LoadDefaultConfig(context.TODO(), config.WithRegion(requestEC2Region()))
```

Why ignoring the error?